pdc/apps/compose/views.py

```diff
@@ -461,11 +463,43 @@ def filter_queryset(self, qs):
         possible to sort unconditionally as get_object() will at some point
         call this method and fail unless it receives a QuerySet instance.)
         """
-        qs = super(ComposeViewSet, self).filter_queryset(qs)
+        qs = super(ComposeViewSet, self).filter_queryset(self._filter_nvras(qs))
         if getattr(self, 'order_queryset', False):
             return sorted(qs)
         return qs
 
+    @staticmethod
+    def _filter_rpm_value(query_params, query_param, rpm_key, qs, q):
+        rpm_value = query_params.get(query_param, None)
+        s = 'variant__variantarch__composerpm__rpm__' + rpm_key
+        if rpm_value is None:
+            return qs, q
+        if len(rpm_value) > 1:
+            condition_list = []
+            for val in rpm_value:
+                condition_list.append(Q(**{s + '__iexact': val}))
+            qs = qs.filter(reduce(operator.or_, condition_list)).distinct()
+            return qs, None
+        if len(rpm_value) == 1:
+            q &= Q(**{s + '__iexact': rpm_value[0]})
```

Comment:
From only read the documentation below, I also thoughtit should work that way @lubomir described. Except I thought the priority will take effect if multiple fields have multiple filter value. It looks straightforward and accord to intuition.

The current implementation is filter according to priority until the end or the first occurence of multiple filter values in one filed,right? I think it's a little complex and hard to understand.